agent/api/ecsclient/client.go

```diff
@@ -279,6 +279,52 @@ func (client *APIECSClient) getCustomAttributes() []*ecs.Attribute {
 	return attributes
 }
 
+func (client *APIECSClient) buildContainerStateChangePayload(change api.ContainerStateChange) *ecs.ContainerStateChange {
+	statechange := &ecs.ContainerStateChange{
+		ContainerName: &change.ContainerName,
+	}
+
+	if change.Reason != "" {
+		if len(change.Reason) > ecsMaxReasonLength {
+			trimmed := change.Reason[0:ecsMaxReasonLength]
+			statechange.Reason = &trimmed
+		} else {
+			statechange.Reason = &change.Reason
+		}
+	}
+	stat := change.Status.String()
+	if stat == "DEAD" {
```

Comment:
i'm not sure tbh. the check was in the original SubmitContainerStateChange code. although api.ContainerStatus doesnt have a "DEAD" state, so we may be good to remove this. @aaithal any ideas?